x-pack/plugins/timelines/public/components/t_grid/standalone/index.tsx

```diff
@@ -328,6 +324,10 @@ const TGridStandaloneComponent: React.FC<TGridStandaloneProps> = ({
   return (
     <InspectButtonContainer data-test-subj="events-viewer-panel">
       <AlertsTableWrapper>
+        {/* Only show the table-spanning loading indicator for the initial fetch of the data.
+            Subsequent fetches (e.g. for pagination) will show a small loading indicator on
+            top of the table and the tabke will display the current page until the next page
+            is fetched. This prevents a flicker when paginating. */}
         {isFirstUpdate.current && <TGridLoading />}
```

Comment:
We could have used existence of data as well to check if it is the first load, no?.  The will also ensure if the it is not the first load i.e. 3 or 4th and there is not data available, it will still show `TGridLoading` component

For example,

```jsx
{!data && <TGridLoading />}
```

Reply:
@logeekal nice catch, wdyt about this? https://github.com/elastic/kibana/pull/139377/commits/5a82383cabc6378066062ae4959b89153770b818

Reply:
Perfect 💯. I guess only comment needs to reworded according to new changes.

Rest of the changes look good as well. I will just locally test once.

Reply:
imo the gist of the comment is still the same. Which part would you like to have changed?

Reply:
It is simply a nit : comments says the table-spanning loading indicator is only for the initial fetch of the data but it is actually whenever the table is empty.

Reply:
@logeekal Ah yes, sth like this? https://github.com/elastic/kibana/pull/139377/commits/5054714ac340112a791304a1971bf710bf50886a